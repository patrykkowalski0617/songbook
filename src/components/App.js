import React, { Component } from "react";
import { Header } from "./header/";
import { Main } from "./main/";
import { Footer } from "./footer/";
import { connect } from "react-redux";
import { tutorialDeactivate, logIn } from "../redux/actions";
import Spotify from "spotify-web-api-js";

const spotifyApi = new Spotify();

class App extends Component {
    constructor(props) {
        super(props);

        const params = this.getHashParams();

        const token = params.access_token;
        if (token) {
            spotifyApi.setAccessToken(token);
            spotifyApi
                .getMe()
                .then(data => {
                    this.props.logIn(token ? data : false);
                })
                .catch(err => console.log(err));
        }
        this.state = { userImg: null };
    }

    getHashParams() {
        const hashParams = {};
        let e,
            r = /([^&;=]+)=?([^&;]*)/g,
            q = window.location.hash.substring(1);
        while ((e = r.exec(q))) {
            hashParams[e[1]] = decodeURIComponent(e[2]);
        }

        return hashParams;
    }

    componentWillMount() {
        const { tutorialDeactivate } = this.props;

        const tutorialIsInactive = localStorage.getItem("tutorialIsInactive");
        if (tutorialIsInactive) {
            tutorialDeactivate();
        }
    }

    render() {
        return (
            <div>
                <Header />
                <Main />
                <Footer />
            </div>
        );
    }
}

const mapStateToProps = state => {
    return { redux: state };
};
const mapDispatchToProps = {
    tutorialDeactivate,
    logIn
};
export default connect(mapStateToProps, mapDispatchToProps)(App);
