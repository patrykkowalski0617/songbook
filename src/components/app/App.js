import React, { Component } from "react";
import Header from "../header/Header";
import Main from "../main/Main";
import Footer from "../footer/Footer";
import lyrics_list from "./data/lyrics_list";
import searchClick from "./logic/searchClick";
import getLyricsItems from "./logic/getLyricsItems";
import switchButtonIcon from "./logic/switchButtonIcon";
import getLyricsJson from "./logic/getLyricsJson";
import displayLyricsList from "./logic/displayLyricsList";

class App extends Component {
    lyricsList = getLyricsItems(lyrics_list);

    state = {
        displayLyricsList: false,
        displayHeaderButtons: [true, false],
        displayLyrics: false,
        displayWelcomeInfo: true,
        displayCountdown: false,
        headerFocusedButtounIndex: 0,
        buttonsOnStates: [true, true],
        searchResult: this.lyricsList,
        lyricsData: null
    };

    buttonsData = [
        {
            onIcon: "lyrics-list",
            offIcon: "close",
            onClickHandler: () => this.displayLyricsList()
        },
        {
            onIcon: "play",
            offIcon: "pause",
            onClickHandler: () => {
                this.counter.action.toggle();
                this.setState({ displayCountdown: true });
            }
        }
    ];

    displayLyricsList = displayLyricsList(this);

    getLyricsJson = getLyricsJson(this);

    switchButtonIcon = switchButtonIcon(this);

    searchClick = searchClick(this, this.lyricsList);

    render() {
        return (
            <div>
                <Header
                    displayLyricsList={this.state.displayLyricsList}
                    displayHeaderButtons={this.state.displayHeaderButtons}
                    searchClick={this.searchClick}
                    counter={this.counter}
                    buttonsData={this.buttonsData}
                    buttonsOnStates={this.state.buttonsOnStates}
                    switchButtonIcon={index => this.switchButtonIcon(index)}
                    headerFocusedButtounIndex={
                        this.state.headerFocusedButtounIndex
                    }
                />
                <Main
                    displayLyricsList={this.state.displayLyricsList}
                    searchResult={this.state.searchResult}
                    getLyricsJson={this.getLyricsJson}
                    lyricsData={this.state.lyricsData}
                    counter={this.counter}
                    displayLyrics={this.state.displayLyrics}
                    displayWelcomeInfo={this.state.displayWelcomeInfo}
                    displayCountdown={this.state.displayCountdown}
                />
                <Footer />
            </div>
        );
    }
}

export default App;
