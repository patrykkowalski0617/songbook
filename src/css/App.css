* {
  margin: 0;
  padding: 0;
  -webkit-box-sizing: border-box;
          box-sizing: border-box; }

.container {
  max-width: 700px;
  margin: 0 auto;
  padding: 0.7rem; }

.container-fluid {
  margin: 0 auto;
  padding: 0.7rem; }

*[class*="col-"].vertical-padding {
  padding-top: 0.7rem;
  padding-bottom: 0.7rem; }

.row.vertical-padding {
  padding-top: 0.7rem;
  padding-bottom: 0.7rem; }

section,
article {
  padding: 1rem 0 1.9rem 0; }

.row {
  margin-left: -0.7rem;
  margin-right: -0.7rem;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-flow: row wrap;
          flex-flow: row wrap; }

*[class*="col"] {
  display: block;
  padding: 0 0.7rem; }

.col-1 {
  max-width: 8.33%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 8.33%;
          flex: 0 0 8.33%; }

.col-2 {
  max-width: 16.66%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 16.66%;
          flex: 0 0 16.66%; }

.col-3 {
  max-width: 25%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 25%;
          flex: 0 0 25%; }

.col-4 {
  max-width: 33.33%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 33.33%;
          flex: 0 0 33.33%; }

.col-5 {
  max-width: 41.66%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 41.66%;
          flex: 0 0 41.66%; }

.col-6 {
  max-width: 50%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 50%;
          flex: 0 0 50%; }

.col-7 {
  max-width: 58.33%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 58.33%;
          flex: 0 0 58.33%; }

.col-8 {
  max-width: 66.66%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 66.66%;
          flex: 0 0 66.66%; }

.col-9 {
  max-width: 75%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 75%;
          flex: 0 0 75%; }

.col-10 {
  max-width: 83.33%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 83.33%;
          flex: 0 0 83.33%; }

.col-11 {
  max-width: 91.66%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 91.66%;
          flex: 0 0 91.66%; }

.col-12 {
  max-width: 100%;
  -webkit-box-flex: 0;
      -ms-flex: 0 0 100%;
          flex: 0 0 100%; }

.off-0 {
  margin-left: 0; }

.off-1 {
  margin-left: 8.33%; }

.off-2 {
  margin-left: 16.66%; }

.off-3 {
  margin-left: 25%; }

.off-4 {
  margin-left: 33.33%; }

.off-5 {
  margin-left: 41.66%; }

.off-6 {
  margin-left: 50%; }

.off-7 {
  margin-left: 58.33%; }

.off-8 {
  margin-left: 66.66%; }

.off-9 {
  margin-left: 75%; }

.off-10 {
  margin-left: 83.33%; }

.off-11 {
  margin-left: 91.66%; }

.off-12 {
  margin-left: 100%; }

@media (min-width: 320px) {
  .col-s-1 {
    max-width: 8.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 8.33%;
            flex: 0 0 8.33%; }

  .col-s-2 {
    max-width: 16.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 16.66%;
            flex: 0 0 16.66%; }

  .col-s-3 {
    max-width: 25%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 25%;
            flex: 0 0 25%; }

  .col-s-4 {
    max-width: 33.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 33.33%;
            flex: 0 0 33.33%; }

  .col-s-5 {
    max-width: 41.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 41.66%;
            flex: 0 0 41.66%; }

  .col-s-6 {
    max-width: 50%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 50%;
            flex: 0 0 50%; }

  .col-s-7 {
    max-width: 58.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 58.33%;
            flex: 0 0 58.33%; }

  .col-s-8 {
    max-width: 66.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 66.66%;
            flex: 0 0 66.66%; }

  .col-s-9 {
    max-width: 75%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 75%;
            flex: 0 0 75%; }

  .col-s-10 {
    max-width: 83.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 83.33%;
            flex: 0 0 83.33%; }

  .col-s-11 {
    max-width: 91.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 91.66%;
            flex: 0 0 91.66%; }

  .col-s-12 {
    max-width: 100%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 100%;
            flex: 0 0 100%; }

  .off-s-0 {
    margin-left: 0; }

  .off-s-1 {
    margin-left: 8.33%; }

  .off-s-2 {
    margin-left: 16.66%; }

  .off-s-3 {
    margin-left: 25%; }

  .off-s-4 {
    margin-left: 33.33%; }

  .off-s-5 {
    margin-left: 41.66%; }

  .off-s-6 {
    margin-left: 50%; }

  .off-s-7 {
    margin-left: 58.33%; }

  .off-s-8 {
    margin-left: 66.66%; }

  .off-s-9 {
    margin-left: 75%; }

  .off-s-10 {
    margin-left: 83.33%; }

  .off-s-11 {
    margin-left: 91.66%; }

  .off-s-12 {
    margin-left: 100%; } }
@media (min-width: 450px) {
  .col-m-1 {
    max-width: 8.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 8.33%;
            flex: 0 0 8.33%; }

  .col-m-2 {
    max-width: 16.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 16.66%;
            flex: 0 0 16.66%; }

  .col-m-3 {
    max-width: 25%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 25%;
            flex: 0 0 25%; }

  .col-m-4 {
    max-width: 33.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 33.33%;
            flex: 0 0 33.33%; }

  .col-m-5 {
    max-width: 41.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 41.66%;
            flex: 0 0 41.66%; }

  .col-m-6 {
    max-width: 50%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 50%;
            flex: 0 0 50%; }

  .col-m-7 {
    max-width: 58.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 58.33%;
            flex: 0 0 58.33%; }

  .col-m-8 {
    max-width: 66.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 66.66%;
            flex: 0 0 66.66%; }

  .col-m-9 {
    max-width: 75%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 75%;
            flex: 0 0 75%; }

  .col-m-10 {
    max-width: 83.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 83.33%;
            flex: 0 0 83.33%; }

  .col-m-11 {
    max-width: 91.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 91.66%;
            flex: 0 0 91.66%; }

  .col-m-12 {
    max-width: 100%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 100%;
            flex: 0 0 100%; }

  .off-m-0 {
    margin-left: 0; }

  .off-m-1 {
    margin-left: 8.33%; }

  .off-m-2 {
    margin-left: 16.66%; }

  .off-m-3 {
    margin-left: 25%; }

  .off-m-4 {
    margin-left: 33.33%; }

  .off-m-5 {
    margin-left: 41.66%; }

  .off-m-6 {
    margin-left: 50%; }

  .off-m-7 {
    margin-left: 58.33%; }

  .off-m-8 {
    margin-left: 66.66%; }

  .off-m-9 {
    margin-left: 75%; }

  .off-m-10 {
    margin-left: 83.33%; }

  .off-m-11 {
    margin-left: 91.66%; }

  .off-m-12 {
    margin-left: 100%; } }
@media (min-width: 580px) {
  .col-l-1 {
    max-width: 8.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 8.33%;
            flex: 0 0 8.33%; }

  .col-l-2 {
    max-width: 16.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 16.66%;
            flex: 0 0 16.66%; }

  .col-l-3 {
    max-width: 25%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 25%;
            flex: 0 0 25%; }

  .col-l-4 {
    max-width: 33.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 33.33%;
            flex: 0 0 33.33%; }

  .col-l-5 {
    max-width: 41.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 41.66%;
            flex: 0 0 41.66%; }

  .col-l-6 {
    max-width: 50%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 50%;
            flex: 0 0 50%; }

  .col-l-7 {
    max-width: 58.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 58.33%;
            flex: 0 0 58.33%; }

  .col-l-8 {
    max-width: 66.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 66.66%;
            flex: 0 0 66.66%; }

  .col-l-9 {
    max-width: 75%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 75%;
            flex: 0 0 75%; }

  .col-l-10 {
    max-width: 83.33%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 83.33%;
            flex: 0 0 83.33%; }

  .col-l-11 {
    max-width: 91.66%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 91.66%;
            flex: 0 0 91.66%; }

  .col-l-12 {
    max-width: 100%;
    -webkit-box-flex: 0;
        -ms-flex: 0 0 100%;
            flex: 0 0 100%; }

  .off-l-0 {
    margin-left: 0; }

  .off-l-1 {
    margin-left: 8.33%; }

  .off-l-2 {
    margin-left: 16.66%; }

  .off-l-3 {
    margin-left: 25%; }

  .off-l-4 {
    margin-left: 33.33%; }

  .off-l-5 {
    margin-left: 41.66%; }

  .off-l-6 {
    margin-left: 50%; }

  .off-l-7 {
    margin-left: 58.33%; }

  .off-l-8 {
    margin-left: 66.66%; }

  .off-l-9 {
    margin-left: 75%; }

  .off-l-10 {
    margin-left: 83.33%; }

  .off-l-11 {
    margin-left: 91.66%; }

  .off-l-12 {
    margin-left: 100%; } }
* {
  -webkit-box-sizing: border-box;
          box-sizing: border-box; }

body {
  margin: 0;
  padding: 0;
  font-family: "Roboto", sans-serif;
  font-weight: 400;
  font-size: 15px;
  background-color: #000000;
  color: #ffffff;
  overflow: hidden; }

.header,
.footer {
  width: 100%;
  background-color: #000000;
  color: #ffffff; }

a:link,
a:visited,
a:active,
a:hover {
  color: inherit; }

a {
  text-decoration: none; }

ul,
ol {
  list-style-type: none; }

.circle-input,
.bar-input {
  height: 40px;
  border-radius: 20px;
  border: none;
  background-color: #ffffff;
  color: #000000; }

.circle-input {
  width: 40px; }

.bar-input {
  padding-left: 20px;
  padding-right: 20px; }

input:focus,
button:focus,
a:focus {
  outline: none;
  -webkit-box-shadow: #6f807d 0px 0px 0px 2px;
          box-shadow: #6f807d 0px 0px 0px 2px;
  background-color: #defffa;
  z-index: 1; }
