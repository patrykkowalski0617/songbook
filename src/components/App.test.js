import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import store from "../redux/store";

describe("Examining rendering", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<App store={store} />);
        expect(wrapper).toMatchSnapshot();
    });
});
