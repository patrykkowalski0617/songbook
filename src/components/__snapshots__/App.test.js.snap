// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examining rendering renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <App
        counterSetScrollDelay={[Function]}
        redux={
          Object {
            "colorSchemeNo": 1,
            "counterIsRun": false,
            "counterIterationNumber": 0,
            "counterScrollDelay": 1,
            "displayLyricsList": false,
            "lyricsData": null,
            "lyricsIsLastBarMarked": false,
            "searchResult": null,
            "searchedValue": "",
            "songTiming": 0,
            "tutorialIsInactive": false,
            "tutorialStep": 0,
          }
        }
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        tutorialDeactivate={[Function]}
      />,
      "value": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "counterSetScrollDelay": [Function],
        "redux": Object {
          "colorSchemeNo": 1,
          "counterIsRun": false,
          "counterIterationNumber": 0,
          "counterScrollDelay": 1,
          "displayLyricsList": false,
          "lyricsData": null,
          "lyricsIsLastBarMarked": false,
          "searchResult": null,
          "searchedValue": "",
          "songTiming": 0,
          "tutorialIsInactive": false,
          "tutorialStep": 0,
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "tutorialDeactivate": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <App
          counterSetScrollDelay={[Function]}
          redux={
            Object {
              "colorSchemeNo": 1,
              "counterIsRun": false,
              "counterIterationNumber": 0,
              "counterScrollDelay": 1,
              "displayLyricsList": false,
              "lyricsData": null,
              "lyricsIsLastBarMarked": false,
              "searchResult": null,
              "searchedValue": "",
              "songTiming": 0,
              "tutorialIsInactive": false,
              "tutorialStep": 0,
            }
          }
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          tutorialDeactivate={[Function]}
        />,
        "value": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "counterSetScrollDelay": [Function],
          "redux": Object {
            "colorSchemeNo": 1,
            "counterIsRun": false,
            "counterIterationNumber": 0,
            "counterScrollDelay": 1,
            "displayLyricsList": false,
            "lyricsData": null,
            "lyricsIsLastBarMarked": false,
            "searchResult": null,
            "searchedValue": "",
            "songTiming": 0,
            "tutorialIsInactive": false,
            "tutorialStep": 0,
          },
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "tutorialDeactivate": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
